Backend/profile/profile.go

```diff
@@ -154,22 +162,143 @@ func FindUserByName(firstName string, lastName string) []models.User {
 	return returnList
 }
 
+
+func EditName(username string, newFirst string, newLast string) error{
+
+	// Some checks included here to decide on editing first name, last name, or both. 
+	if len(newLast) > 0 && len(newFirst) > 0{
+
+		query := "UPDATE users SET first_name = ?, last_name = ? username= ?"
+		_, err := db.Exec(query,newFirst,newLast,username)
+
+		if err != nil {
+			return err
+		}
+
+	}else if len(newLast) > 0{
+
+		query := "UPDATE users SET last_name = ? username = ?"
+		_, err := db.Exec(query,newLast,username)
+
+		if err != nil {
+			return err
+		}
+
+	}else{
+
+		query := "UPDATE users SET first_name = ? username = ?"
+		_, err := db.Exec(query,newFirst,username)
+
+		if err != nil {
+	
+		}
+
+	}
+
+	return nil
+}
+
+func EditBio(username string, newBio string) error{
+
+	if len(newBio) > 0{
+
+		query := "UPDATE users SET bio = ? WHERE username = ?"
+		_, err := db.Exec(query,newBio,username)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func EditUsername(username string, newUsername string) error{
+
+	if len(newUsername) > 0{
+
+		query := "UPDATE users SET username = ? WHERE username = ?"
+		_, err := db.Exec(query,newUsername,username)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DeleteUser(username string) error{
+
+	//Beginning transactions for the database, so they can be rolled back if an error occurs midway.
+	txn, err := db.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	// deferring the function to either commit the transactions, or roll them back depending on if an error is thrown. 
+    defer func() {
+        if err != nil {
+            txn.Rollback()
+        } else {
+            err = txn.Commit()
+        }
+    }()
+	
+	var userID int
+	// getting username to delete from Friends table
+	err = txn.QueryRow("SELECT id FROM users WHERE username = ?",username).Scan(&userID)
+
+	// removing from reactions table
+	_, err = txn.Exec("DELETE FROM reactions WHERE user_id = ?",userID)
+	if err != nil {
+		return err
+	}
+
+	// removing from replies table
+	_, err = txn.Exec("DELETE FROM replies WHERE user_id = ?",userID)
+
+	if err != nil {
+		return err
+	}
+
+	// removing from posts table
+	_, err = txn.Exec("DELETE FROM posts WHERE user_id = ?",userID)
+
+	if err != nil {
+		return err
+	}
+
+	// removing from friends table
+	_, err = txn.Exec("DELETE FROM friends WHERE user_id = ?",username)
```

Comment:
would we also need to delete OR friend_id=? since we have it twice for each friendship?